ingest.py

```diff
@@ -91,49 +93,63 @@ def load_single_document(file_path: str) -> Document:
     raise ValueError(f"Unsupported file extension '{ext}'")
 
 
-def load_documents(source_dir: str, ignored_files: List[str] = []) -> List[Document]:
+def load_documents(source_dir: str, ignored_files: Optional[List[str]] = None) -> List[Document]:
     """
     Loads all documents from the source documents directory, ignoring specified files
     """
-    all_files = []
-    for ext in LOADER_MAPPING:
-        all_files.extend(
-            glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
-        )
-    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
+    filtered_files = []
+
+    ext_pattern = re.compile(
+        r"({})$".format("|".join(re.escape(ext) for ext in LOADER_MAPPING)),
+        re.IGNORECASE
+    )
+
+    for root, _, files in os.walk(source_dir):
+        for file in files:
+            if ext_pattern.search(file):
+                file_path = os.path.join(root, file)
+                if file_path not in ignored_files:
+                    filtered_files.append(file_path)
 
     with Pool(processes=os.cpu_count()) as pool:
         results = []
         with tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) as pbar:
-            for i, doc in enumerate(pool.imap_unordered(load_single_document, filtered_files)):
+            for _, doc in enumerate(pool.imap_unordered(load_single_document, filtered_files)):
                 results.append(doc)
                 pbar.update()
 
     return results
 
-def process_documents(ignored_files: List[str] = []) -> List[Document]:
+def process_documents(ignored_files: Optional[List[str]] = None) -> List[Document]:
     """
     Load documents and split in chunks
     """
+
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
     print(f"Loading documents from {source_directory}")
     documents = load_documents(source_directory, ignored_files)
     if not documents:
         print("No new documents to load")
-        exit(0)
+        sys.exit(0)
     print(f"Loaded {len(documents)} new documents from {source_directory}")
     text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)
     texts = text_splitter.split_documents(documents)
     print(f"Split into {len(texts)} chunks of text (max. {chunk_size} tokens each)")
     return texts
 
-def does_vectorstore_exist(persist_directory: str) -> bool:
+def does_vectorstore_exist(persist_path: str) -> bool:
```

Comment:
I suppose you changed the name because it shadows the global one. But note that since it's a global, we don't need to pass it to this function at all. So you could just remove the parameter and leave the name unchanged.

That being said, since using globals is not great and eventually I think we'll change that, this change still makes sense.

Reply:
It's not good practice to use globals - I try to use globals only when they are real constants.   There's generally a reason down the road to not use globals.